test/src/waittask.spec.ts

```diff
@@ -478,113 +479,202 @@ describe('waittask specs', function () {
       await waitForSelector;
       expect(boxFound).toBe(true);
     });
-    it('should wait for visible', async () => {
+    it('should wait for element to be visible (display)', async () => {
       const {page} = getTestState();
 
-      let divFound = false;
-      const waitForSelector = page
-        .waitForSelector('div', {visible: true})
-        .then(() => {
-          return (divFound = true);
-        });
-      await page.setContent(
-        `<div style='display: none; visibility: hidden;'>1</div>`
-      );
-      expect(divFound).toBe(false);
```

Comment:
@OrKoN I've used timeouts since `divFound` isn't necessarily changed right after setting content. To be specific, it could be the case that waitForSelector worked but the promise has yet to complete (I've tested this).